### PR TITLE
add internal/compress/lz4_option test

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

- Add test for `internal/compress/lz4_option`
- Update comments for each external func/method/var in `internal/compress/lz4_option`

- Note: I could not cover the error pattern test in `WithLZ4Gob`, because `gobCompressor` is an empty struct and has no `options`.

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/compress/lz4_option_test.go [hunk 1]
```diff
@@ -18,196 +18,142 @@
...
+				args: args{
+					level: -1,
+				},
+				want: want{
+					obj: &T{
+						compressionLevel: -1,
+					},
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline internal/compress/lz4_option_test.go:136]:
```suggestion
		{
			name: "set success when level is less than 0.",
			args: args{
				level: -1,
			},
			want: want{
				obj: &T{
					compressionLevel: -1,
				},
			},
		},
```

File: internal/compress/lz4_option_test.go [hunk 2]
```diff
@@ -18,196 +18,142 @@
...
+		func() test {
```
Comment (kevindiu) [inline internal/compress/lz4_option_test.go:137]: same

File: internal/compress/lz4_option_test.go [hunk 3]
```diff
@@ -18,196 +18,142 @@
...
+			return test{
+				name: "set success when level is nil.",
+				want: want{
+					obj: new(T),
+				},
+			}
+		}(),
+		func() test {
```
Comment (kevindiu) [inline internal/compress/lz4_option_test.go:145]: same

File: internal/compress/lz4_option_test.go [hunk 4]
```diff
@@ -18,196 +18,136 @@
...
+				obj: new(T),
+			},
+		},
+		{
+			name: "return error when level is more than 1.",
+			args: args{
+				level: 1,
+			},
+			want: want{
+				obj: new(T),
```
Comment (kevindiu) [inline internal/compress/lz4_option_test.go:147]:
```suggestion
```
